src/nnsight/tracing/protocols.py

```diff
@@ -490,3 +511,163 @@ def execute(cls, node: Node) -> None:
     def set(cls, node: Node, value: Any) -> None:
 
         node.args[0] = value
+
+class ConditionalProtocol(Protocol):
+    """ Protocol operating as a conditional statement. 
+    Uses the ConditionalManager attachment to handle all visited Conditional contexts within a single Intervention Graph.
+    Evaluates the condition value of the Conditional as a boolean.
+
+    Example:
+
+        Setup:
+            .. code-block:: python
+                import torch
+                from collections import OrderedDict
+
+                input_size = 5
+                hidden_dims = 10
+                output_size = 2
+
+                model = nn.Sequential(OrderedDict([
+                    ('layer1', torch.nn.Linear(input_size, hidden_dims)),
+                    ('layer2', torch.nn.Linear(hidden_dims, output_size)),
+                ]))
+
+                input = torch.rand((1, input_size))    å
+
+        Ex 1: The .save() on the model output will only be executed if the condition (x > 0) is evaluated to True.    
+    
+        .. code-block:: python
+            with model.trace(input) as tracer:
+                num = tracer.apply(int, 5)
+                with x > 0:
+                    out = model.output.save()
+
+        Ex 2: The condition is a tensor boolean operation on the Envoy's output InterventionProxy.
+
+        .. code-block:: python
+            with model.trace(input) as tracer:
+                l1_out = model.layer1.output
+                with l1_out[:, 0] > 0:
+                    out = model.output.save()
+    """
+
+    attachment_name = "nnsight_conditional_manager"
+
+    @classmethod
+    def add(cls, node: "Node") -> "InterventionProxy":
+
+        return node.graph.create(target=cls, proxy_value=True, args=[node])
+    
+    @classmethod
+    def execute(cls, node: "Node") -> None:
+        """ Evaluate the node condition to a boolean. 
+        
+        Args:
+            node (Node): ConditionalProtocol node.
+        """
+        
+        cond_value = Node.prepare_inputs(node.args[0])
+        if cond_value:
+            # cond_value is True
+            node.set_value(True)
+            return
+            
+        def update_conditioned_nodes(conditioned_node: "Node") -> None:
+            """ Recursively decrement the remaining listeners count of all the dependencies of conditioned nodes.
+            
+            Args:
+                - conditioned_node (Node): Conditioned Node
+            """
+            for listener in conditioned_node.listeners:
```

Comment:
The recursiveness is required for handling nested `Conditional` contexts. Especially, because nested conditional nodes might not have a direct dependency-listener relationship (due to the InterventionProxy as a conditional syntax design. See unit test: `test_false_conditional_protocol` in `test_tiny.py`)

Reply:
Yes we should probably destroy redundant nodes! I'll push that now. Should we also check if the nodes' value is set?